CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua

```diff
@@ -9,42 +9,164 @@
 
+  self._isMaximized = isMaximized
+
   self._frame = Instance.new("Frame")
   self._frame.Name = "PS_AnnotatedGraph"
+  self._frame.BackgroundTransparency = 1.0
+  
+  self._topLabel = Instance.new("TextLabel")
+  self._topLabel.Name = "PS_TopAxisLabel"
+  StatsUtils.StyleTextWidget(self._topLabel)
+  self._topLabel.Parent = self._frame
+  self._topLabel.TextXAlignment = Enum.TextXAlignment.Left
+  self._topLabel.TextYAlignment = Enum.TextYAlignment.Top
+  
+  self._midLabel = Instance.new("TextLabel")
+  self._midLabel.Name = "PS_MidAxisLabel"
+  StatsUtils.StyleTextWidget(self._midLabel)
+  self._midLabel.Parent = self._frame
+  self._midLabel.TextXAlignment = Enum.TextXAlignment.Left
+  self._midLabel.TextYAlignment = Enum.TextYAlignment.Center
+  
+  self._bottomLabel = Instance.new("TextLabel")
+  self._bottomLabel.Name = "PS_BottomAxisLabel"
+  StatsUtils.StyleTextWidget(self._bottomLabel)
+  self._bottomLabel.Parent = self._frame
+  self._bottomLabel.TextXAlignment = Enum.TextXAlignment.Left
+  self._bottomLabel.TextYAlignment = Enum.TextYAlignment.Bottom
 
-  StatsUtils.StyleFrame(self._frame)
+  
+  self._graph = BarGraphClass.new()
+
+  self:_layoutElements()
 
   return self
 end
 
+
+function StatsAnnotatedGraphClass:_layoutElements()
+  local labelWidth
+  if (self._isMaximized) then
+    labelWidth = LabelXWidth
+    
+    self._topLabel.Visible = true
+    self._midLabel.Visible = true
+    self._bottomLabel.Visible = true
+  else
+    labelWidth = 0
+    
+    self._topLabel.Visible = false
+    self._midLabel.Visible = false
+    self._bottomLabel.Visible = false
+  end
+  
+  local GraphFramePosition = UDim2.new(0, Margin, 0, Margin)
+  local GraphFrameSize = UDim2.new(1, -(2 * Margin + labelWidth), 1, -2 * Margin)
+
+  local TopLabelFramePosition = UDim2.new(1, -(Margin + labelWidth), 0, Margin)
+  local TopLabelFrameSize = UDim2.new(0, labelWidth, 0.333, -2 * Margin)
+  local MidLabelFramePosition = UDim2.new(1, -(Margin + labelWidth), 0.333, Margin)
+  local MidLabelFrameSize = UDim2.new(0, labelWidth, 0.333, -2 * Margin)
+  local BottomLabelFramePosition = UDim2.new(1, -(Margin + labelWidth), 0.666, Margin)
+  local BottomLabelFrameSize = UDim2.new(0, labelWidth, 0.333, -2 * Margin)
+  
+  self._topLabel.Size = TopLabelFrameSize
+  self._topLabel.Position = TopLabelFramePosition
+  self._midLabel.Size = MidLabelFrameSize
+  self._midLabel.Position = MidLabelFramePosition
+  self._bottomLabel.Size = BottomLabelFrameSize
+  self._bottomLabel.Position = BottomLabelFramePosition
+  
+  self._graph:PlaceInParent(self._frame, GraphFrameSize, GraphFramePosition)
+end
+
 function StatsAnnotatedGraphClass:PlaceInParent(parent, size, position) 
   self._frame.Position = position
   self._frame.Size = size
   self._frame.Parent = parent
 end
 
 function StatsAnnotatedGraphClass:SetValues(values)
+  local axisMax = self:_calculateAxisMax(values)
+  self._graph:Render(values, axisMax)
+  
   -- FIXME(dbanks)
```

Comment:
Yes.  Right now I just read raw numbers out of stats and render them to screen.  But things are more readable if I can put in units/formatting based on stat type (so Network stats look like 1.2 Kb/s instead of 1223.002)